internal/cmd/log-file-metric-exporter/main.go

```diff
@@ -0,0 +1,153 @@
+        "github.com/prometheus/client_golang/prometheus"
+        "github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+
+
+
+var (
+ verbosity int = 0
+ //Reference regexp https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/blob/master/lib/fluent/plugin/filter_kubernetes_metadata.rb#L56, https://github.com/kubernetes/kubernetes/blob/release-1.6/pkg/kubelet/dockertools/docker.go
+ k8logfilepathPattern string = `.var.log.containers.([a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(.+)_(.+)-([a-z0-9]{64})\.log$`
+	//compile k8 logfilepathname pattern 
+ kubernetesregexpCompiled = regexp.MustCompile(k8logfilepathPattern)
+)
+
+
+
+
+type FileWatcher struct {
+        watcher *symnotify.Watcher
+        metrics *prometheus.CounterVec
+        sizes  map[string]float64
+        added  map[string]bool
+}
+
+
+func (w *FileWatcher) Update(path string, namespace string, podname string, containername string) error {
+        counter, err := w.metrics.GetMetricWithLabelValues(path,namespace,podname,containername)
+        if err != nil {
+                return err
+        }
+        stat, err := os.Stat(path)
+        if err != nil {
+                return err
+        }
+        if stat.IsDir() {
+                return nil // Ignore directories
+        }
+        lastSize, size := float64(w.sizes[path]), float64(stat.Size())
+        w.sizes[path] = size
+        var add float64
+        if size > lastSize {
+                // File has grown, add the difference to the counter.
+                add = size - lastSize
+        } else if size < lastSize {
+                // File truncated, starting over. Add the size.
+                add = size
+        }
+        log.V(2).Info("For logfile in...", "path",path,"lastsize",lastSize,"currentsize",size,"addedbytes",add)
+	counter.Add(add)
+        return nil
+}
+
+func (w *FileWatcher) Watch( ) {
+
+        
+	for {
+		//All logfiles with containername are added to the watcher
+		//write event for these logfiles are being watched
+		//create event gets issued for all new logfiles appear under logfilepathname /var/log/containers/
+		//For the cases new log files added, old files moved, old files deleted, you need to add/remove them from watcher as whole dir added to the watcher
+		//For new log files added write event is not getting issued
+		
+                e, err := w.watcher.Event()
+                if err != nil {
+		log.Error(err, "Watcher.Event returning err")
+		os.Exit(1)
+		}
+
+                log.V(2).Info("Events notified for...", "e.Name",e.Name,"Event",e.Op)
+
+		//Get namespace, podname, containername from e.Name - log file path
+		//filename := strings.Trim(e.Name,"/var/log/containers/")
+
+		_,filename := path.Split(e.Name)
+                filename = strings.TrimRight(filename,".log")
+                log.V(2).Info("file name after trimming","filename", filename)
+		submatches := kubernetesregexpCompiled.FindAllString(e.Name, -1)
```

Comment:
I'm missing what is happening here.  Utilizing regex, you should be able to write a set of capture groups to get all the needed information without  needing to do any string manipulation.  This looks like you are attempting to combine both string manipulation and regex which negates using one or the other